### PR TITLE
Add report metadata

### PR DESCRIPTION
**Pull request by AleksandarSavchev**
**What this PR does / why we need it**:
This PR creates a new `metadata` field in the Diki config file. The value of this field will be present in the json summary reports as report metadata. This metadata can be viewed in the html report as well, in merged html reports the metadata is only previewed if all merged reports have identical metadata.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
Report metadata can now be added by setting the `metadata` field in the Diki config file.
```

### CODE REVIEW

File: cmd/diki/app/app.go [hunk 1]
```diff
@@ -339,6 +339,9 @@ func runCmd(ctx context.Context, providerCreateFuncs map[string]provider.Provide
 			if dikiConfig.Output != nil && len(dikiConfig.Output.MinStatus) > 0 {
 				reportOpts = append(reportOpts, report.MinStatus(dikiConfig.Output.MinStatus))
 			}
+			if dikiConfig.Metadata != nil && len(dikiConfig.Metadata) > 0 {
```
Comment (dimityrmirchev) [inline cmd/diki/app/app.go:342]:
```suggestion
			if len(dikiConfig.Metadata) > 0 {
```

File: cmd/diki/app/app.go [hunk 2]
```diff
@@ -364,6 +367,9 @@ func runCmd(ctx context.Context, providerCreateFuncs map[string]provider.Provide
...
```
Comment (dimityrmirchev) [inline cmd/diki/app/app.go:370]:
```suggestion
			if len(dikiConfig.Metadata) > 0 {
```

File: cmd/diki/app/app.go [hunk 3]
```diff
@@ -387,6 +393,9 @@ func runCmd(ctx context.Context, providerCreateFuncs map[string]provider.Provide
...
```
Comment (dimityrmirchev) [inline cmd/diki/app/app.go:396]:
```suggestion
			if len(dikiConfig.Metadata) > 0 {
```

File: pkg/report/render.go [hunk 4]
```diff
@@ -46,12 +49,20 @@ func NewHTMLRenderer() (*HTMLRenderer, error) {
 		_, ok := m[k]
 		return ok
 	}
+	yamlFormat := func(m map[string]any) string {
+		yaml, err := yaml.Marshal(m)
+		if err != nil {
+			log.Println(err)
```
Comment (dimityrmirchev) [inline pkg/report/render.go:55]:
```suggestion
			return err.String()
```

Let's show the error in the report if there is such

File: pkg/report/report.go [hunk 5]
```diff
@@ -79,6 +81,14 @@ func (ms MinStatus) ApplyToReport(opts *ReportOptions) {
 	}
 }
 
+// Metadata is additional report values.
+type Metadata map[string]any
+
+// ApplyToReport implements ReportOption.
+func (md Metadata) ApplyToReport(opts *ReportOptions) {
+	opts.Metadata = md
```
Comment (dimityrmirchev) [inline pkg/report/report.go:89]: Can't we use maps.Clone here?

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
A few minor comments, other looks good. Can you please attach an image of how the new changes look in the html report?

**Comment by AleksandarSavchev** [comment]
> Can you please attach an image of how the new changes look in the html report?

Yes, here are two examples:
image
image

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**